src/components/TodoStore/TodoStore.tsx

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import React, { useEffect, useReducer } from 'react';
+
+import { getTodos } from '../../api/todos';
+
+import { Filter } from '../../types/Filter';
+import { Todo } from '../../types/Todo';
+import { TodoError } from '../../types/TodoError';
+
+const USER_ID = 11722;
+
+enum ReducerActions {
+  LoadTodos = 'loadTodos',
+  AddTodo = 'addTodo',
+  UpdateTodo = 'updateTodo',
+  DeleteTodo = 'deleteTodo',
+  FilterTodo = 'filterTodo',
+  AddTodoError = 'addTodoError',
+  ClearTodoErrors = 'clearTodoErrors',
+}
+
+interface Action {
+  type: ReducerActions;
+  payload?: any;
+}
+
+interface State {
+  initialTodos: Todo[];
+  visibleTodos: Todo[];
+  filter: Filter;
+  todoError: TodoError | null;
+}
+
+function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case ReducerActions.LoadTodos:
+      return {
+        ...state,
+        initialTodos: action.payload,
+        visibleTodos: action.payload,
+      };
+
+    case ReducerActions.FilterTodo: {
+      const filteredTodos = state.initialTodos.filter(todo => {
+        switch (action.payload) {
+          case Filter.Active:
+            return !todo.completed;
+
+          case Filter.Completed:
+            return todo.completed;
+
+          case Filter.All:
+          default:
+            return true;
+        }
+      });
```

Comment:
Get rid of this action. Just filter in `TodoAppContext` body and store as a variable